satpy/readers/viirs_vgac_l1c_nc.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2009-2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Reading VIIRS VGAC data."""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+
+class VGACFileHandler(BaseFileHandler):
+    """Reader VGAC data."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(VGACFileHandler, self).__init__(
+            filename, filename_info, filetype_info)
+
+        self.engine = "h5netcdf"
+        self._start_time = filename_info['start_time']
+        self._end_time = None
+        self.sensor = 'viirs'
+        self.filename_info = filename_info
+
+    def convert_to_bt(self, data, data_lut, scale_factor):
+        """Convert radances to brightness temperatures."""
+        from scipy import interpolate
+        x = np.arange(0, len(data_lut))
+        y = data_lut
+        func = interpolate.interp1d(x, y)
+        brightness_temperatures = func(data.values / scale_factor)
```

Comment:
Does this preserve the dask array, or does it create a numpy array?

Reply:
It creates a numpy array.

Reply:
something like this might help then:
```
brightness_temperatures = data.map_blocks(func)
```

Reply:
I did not get it to work. Also we might not use this function for the bt-conversion in the end. I think it is correct but I still have no real values (counts, BT-temperatures) to test it with. And it might be good to keep it simple with numpy arrays as others working with this reader are more familiar with numpy. And the way it will be used, mainly to create level1c files, means that the values will be needed for sure (and written to file). So likely no need to calculate them lazy.  So following the dask guide https://docs.dask.org/en/stable/best-practices.html and stop using dask when no longer needed ;).

Reply:
I agree with the dask guide when it comes to small data, but we are talking about a full orbit of data right?
Moreover, I'm also thinking that one import aspect is to keep the data as xr.DataArrays, otherwise we loose the metadata. So here is a proposal:
```python
        scaled_data = data / scale_factor
        brightness_temperatures = xr.DataArray(scaled_data.data.map_blocks(np.interp, xp=x, fp=y), coords=data.coords, attrs=data.attrs)
```
You may want to alter the attrs to have the right units though.

Alternatively, if the data is really small you can do
```python
        scaled_data = data / scale_factor
        brightness_temperatures = xr.DataArray(np.interp(scaled_data, xp=x, fp=y), coords=data.coords, attrs=data.attrs)
```

Note that I'm proposing the use of `np.interp` as an alternative to scipy, as we try to keep the dependency to scipy at a minimum.